File: backend/app/schemas/provider.py
"""Схемы для провайдеров LLM."""

from pydantic import BaseModel


class ModelInfo(BaseModel):
    """Информация о модели."""

    id: str
    name: str
    description: str | None = None
    max_tokens: int = 4096
    supports_streaming: bool = True


class ProviderInfo(BaseModel):
    """Информация о провайдере."""

    id: str
    name: str
    description: str | None = None
    models: list[ModelInfo] = []
    is_available: bool = True


class ProviderStatus(BaseModel):
    """Статус доступности провайдера."""

    provider_id: str
    is_available: bool
    latency_ms: int | None = None
    error: str | None = None


class ProvidersListResponse(BaseModel):
    """Список провайдеров."""

    providers: list[ProviderInfo]

File: frontend/src/types/provider.ts
// Типы для провайдеров

export interface Model {
  id: string;
  name: string;
  description: string | null;
  max_tokens: number;
  supports_streaming: boolean;
}

export interface Provider {
  id: string;
  name: string;
  description: string | null;
  models: Model[];
  is_available: boolean;
}

export interface ProvidersResponse {
  providers: Provider[];
}
